### PR TITLE
Upgrade deps (tests are now failing)

### PR DESCRIPTION
**Pull request by cakoose**

### CODE REVIEW

File: src/__checks__/select.check.ts [hunk 1]
```diff
@@ -42,201 +42,212 @@ const bar = defineTable({
...
-      toSnap(db.select(db.foo.name, db.bar.startDate, db.bar.value).from(db.foo).rightJoin(db.bar)),
-    );
+    }>();
   });
 
   test('should select * and return nullable properties of left side properties when right joining', () => {
-    expectType<{
-      id: string | null;
+    expect(toSnap(db.select(star()).from(db.foo).rightJoin(db.bar))).type.toEqual<{
+      id: unknown;
```
Comment (carterzenk) [inline src/__checks__/select.check.ts:88]: These changes to `unknown` are caught in the newer version of typescript.

The problem is with queries that use `star` with a join, and the resulting columns have overlapping names, so the resulting type can be ambiguous.  In the existing examples, both tables have the same types for these overlapping columns, so this previously used to work.  Now, it appears that typescript is catching this and returning an `unknown` type instead.

When a query is run with overlapping column names, the order of the columns in the select determines which value will be used.  It is (maybe) possible to update the types to take order into account, and overwrite types based on the order, but this seems like a pretty large change to make.

I think that leaving these overlapping names returning `unknown` instead is safer, and can avoid errors from overlapping by mistake.

File: package.json [hunk 2]
```diff
@@ -21,22 +21,23 @@
...
-    "prettier": "^3.0.3",
-    "ts-jest": "^26.5.6",
-    "typescript": "^4.3.5"
+    "@tsd/typescript": "^5.4.3",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^16.18.94",
+    "jest": "^29.7.0",
+    "prettier": "^3.2.5",
+    "ts-jest": "^29.1.2",
+    "tstyche": "^1.1.0",
```
Comment (carterzenk) [inline package.json:30]: We were previously using [tsd-lite](https://github.com/mrazauskas/tsd-lite) along with `jest-runner-tsd` to run the type checks.  I noticed some issues with the type checks along with the typescript upgrade, and it appears that tsd-lite is now deprecated.  They recommend using [tstyche](https://github.com/tstyche/tstyche) instead, so I switched to this and updated the checks.

The only issue with tstyche is that it does not have a jest runner yet, so we have to run these checks with a separate command.  I updated the `package.json` to add a `check` command to run these tests, and updated the github workflow to add this step.

File: src/SelectFn.ts [hunk 3]
```diff
@@ -4,17 +4,17 @@ import { Expression } from './expression';
...
+  S extends Column<infer A2, string, any, boolean, boolean, any>
+    ? A2
+    : S extends Expression<any, boolean, infer A1>
+      ? A1
+      : S extends SelectQuery<infer Columns>
+        ? keyof Columns // This only works if the query has one select clause
+        : never;
+
+export type GetSelectable<C extends Selectable> =
+  C extends ColumnSet<infer Columns> ? Columns : { [K in GetSelectableName<C>]: C };
```
Comment (carterzenk) [inline src/SelectFn.ts:17]: It appears that these formatting changes are coming from the newer version of prettier.

**Comment by carterzenk** [comment]
@cakoose I fixed the tests, notable changes are described in comments above.  Take a look and let me know if these changes look good to you.

**Comment by cakoose** [comment]
Looks good!

Can you create a new PR from this branch so I can approve? (Not a huge deal, but keeps the authorship a bit more accurate.)

**Pull request closed without merging**